### PR TITLE
[SPARK-5195][sql]Update HiveMetastoreCatalog.scala(override the MetastoreRelation's sameresult method only compare databasename and table name)

### PR DESCRIPTION
**Pull request by seayi**
override  the MetastoreRelation's  sameresult method only compare databasename and table name

because in previous :
cache table t1;
select count(_) from t1;
it will read data from memory  but the sql below will not,instead it read from hdfs:
select count(_) from t1 t; 

because cache data is keyed by logical plan and compare with sameResult ,so  when table with alias  the same table 's logicalplan is not the same logical plan with out alias  so modify  the sameresult method only compare databasename and table name

### CODE REVIEW

**Comment by seayi** [comment]
override  the sameresult method only compare databasename and table name

**Comment by seayi** [comment]
i test with hive table,after modify the sameresult method it is ok

**Comment by marmbrus** [comment]
Can you also add a regression test to [`CachedTableSuite`](https://github.com/apache/spark/blob/master/sql/hive/src/test/scala/org/apache/spark/sql/hive/CachedTableSuite.scala)?

**Comment by marmbrus** [comment]
ok to test

File: sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveMetastoreCatalog.scala [hunk 1]
```diff
@@ -427,6 +427,13 @@ private[hive] case class MetastoreRelation
           .getOrElse(sqlContext.defaultSizeInBytes))
     }
   )
+  override def sameResult(plan: LogicalPlan): Boolean = {
+    val new_plan = plan.asInstanceOf[MetastoreRelation];
```
Comment (marmbrus) [inline sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveMetastoreCatalog.scala]: what about when its not a metastore relation?  how about:

```scala
plan match {
  case mr: MetastoreRelation =>
    mr.databaseName == databaseName && mr.tableName == tableName
  case _ => false
}
```

**Comment by marmbrus** [comment]
Finally, please file a JIRA and add it to the PR title.  Thanks!

File: sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveMetastoreCatalog.scala (hunk 1 above)
Comment (seayi) [inline sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveMetastoreCatalog.scala]: ok   i will  add  thanks

**Comment by marmbrus** [comment]
ping :)

**Comment by seayi** [comment]
sorry  i am busy these days   i will  file a JIRA and add it to the PR title and pull a new request  as soon as possibly.

File: sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveMetastoreCatalog.scala [hunk 2]
```diff
@@ -427,6 +427,13 @@ private[hive] case class MetastoreRelation
           .getOrElse(sqlContext.defaultSizeInBytes))
     }
   )
+  override def sameResult(plan: LogicalPlan): Boolean = {
+      plan match {
+           case mr: MetastoreRelation =>
```
Comment (marmbrus) [inline sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveMetastoreCatalog.scala]: nit: 2 spaces for intents.

I'll fix while merging.

**Comment by marmbrus** [comment]
Thanks for fixing this!  Merging to master.

**Comment by marmbrus** [comment]
Ahhh, I just noticed this was not made against master.  Please make all PRs against master in the future.

File: sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveMetastoreCatalog.scala (hunk 2 above)
Comment (seayi) [inline sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveMetastoreCatalog.scala]: thanks, marmbrus   but i don't know how to make  PRs against master,can you tell me? thanks again for your attension

Comment (marmbrus) [inline sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveMetastoreCatalog.scala]: When you open the PR make sure the base branch says `master` not `branch-1.X`.  https://help.github.com/articles/comparing-commits-across-time/#comparing-branches

**Pull request closed without merging**